packages/manager/src/managers/git/GitManager.ts

```diff
@@ -0,0 +1,296 @@
+
+import { UnauthorizedError, UnexpectedDataError } from "../../errors";
+
+import { BaseManager } from "../BaseManager";
+
+import { GitRepo, GitRepoSpcifier, Namespace } from "./types";
+
+type CreateGitHubAuthStateReturnType = {
+	key: string;
+	expiresAt: Date;
+};
+
+type GitManagerFetchOwnersReturnType = Namespace[];
+
+type GitManagerFetchReposReturnType = GitRepo[];
+
+type GitManagerFetchReposArgs = {
+	provider: "gitHub";
+	owner: string;
+	query?: string;
+	page?: number;
+};
+
+type GitManagerFetchLinkedReposArgs = {
+	prismic: {
+		domain: string;
+	};
+};
+
+type GitManagerFetchLinkedReposReturnType = GitRepoSpcifier[];
+
+type GitManagerLinkRepoArgs = {
+	prismic: {
+		domain: string;
+	};
+	git: {
+		provider: "gitHub";
+		owner: string;
+		name: string;
+	};
+};
+
+type GitManagerUnlinkRepoArgs = {
+	prismic: {
+		domain: string;
+	};
+	git: {
+		provider: "gitHub";
+		owner: string;
+		name: string;
+	};
+};
+
+export class GitManager extends BaseManager {
+	async createGitHubAuthState(): Promise<CreateGitHubAuthStateReturnType> {
+		const url = new URL(
+			"./git/github/create-auth-state",
+			API_ENDPOINTS.SliceMachineV1,
+		);
+		const res = await this.#fetch(url);
+
+		if (!res.ok) {
+			switch (res.status) {
+				case 401:
+					throw new UnauthorizedError();
+				default:
+					throw new Error("Failed to create GutHub auth state.");
+			}
+		}
+
+		const json = await res.json();
+		const { value, error } = decode(
+			t.type({
+				key: t.string,
+				expiresAt: tt.DateFromISOString,
+			}),
+			json,
+		);
+
+		if (error) {
+			throw new UnexpectedDataError(
+				`Failed to decode GitHub auth state: ${error.errors.join(", ")}`,
+				{ cause: error },
+			);
+		}
+
+		return value;
+	}
+
+	async fetchOwners(): Promise<GitManagerFetchOwnersReturnType> {
+		const url = new URL("./git/owners", API_ENDPOINTS.SliceMachineV1);
+		const res = await this.#fetch(url);
+
+		if (!res.ok) {
+			switch (res.status) {
+				case 403:
+					throw new UnauthorizedError();
+				default:
+					throw new Error("Failed to fetch owners.");
+			}
+		}
+
+		const json = await res.json();
+		const { value, error } = decode(
+			t.type({
+				owners: t.array(
+					t.type({
+						provider: t.literal("gitHub"),
+						id: t.string,
+						name: t.string,
+						type: t.union([t.literal("user"), t.literal("team"), t.null]),
+					}),
+				),
+			}),
+			json,
+		);
+
+		if (error) {
+			throw new UnexpectedDataError(
+				`Failed to decode owners: ${error.errors.join(", ")}`,
+				{ cause: error },
+			);
+		}
+
+		return value.owners;
+	}
+
+	async fetchRepos(
+		args: GitManagerFetchReposArgs,
+	): Promise<GitManagerFetchReposReturnType> {
+		const url = new URL("./git/repos", API_ENDPOINTS.SliceMachineV1);
+		url.searchParams.set("provider", args.provider);
+		url.searchParams.set("owner", args.owner);
+		if (args.query) {
+			url.searchParams.set("q", args.query);
+		}
+		if (args.page) {
```

Comment:
I think we want that behavior since `page=0` is not valid. However, maybe we need to be stricter, such as only supporting positive numbers.

I changed the condition to:

```ts
if (args.page && args.page > 0) { ... }
```

If you want to enforce boolean checking, do you think we should add that to the ESLint config?